src/services/BlockMinter.ts

```diff
@@ -28,9 +29,11 @@ class BlockMinter {
   @inject(BlockRepository) blockRepository!: BlockRepository;
   @inject(ConsensusDataRepository) consensusDataRepository!: ConsensusDataRepository;
   @inject(MultichainArchitectureDetector) multichainArchitectureDetector!: MultichainArchitectureDetector;
+  @inject(BalanceService) balanceService!: BalanceService;
   @inject('Settings') settings!: Settings;
 
   async apply(): Promise<void> {
+    await this.balanceService.checkBalanceIsEnough(this.blockchain, this.settings.blockchain.masterChain.chainId);
```

Comment:
ah, that way when we have AVAX but we don't have BSC we will not run it at all. So turns out, either we need to check for all supported chains or we can not check it at all. let me think about it.